src/dom/Null.ts

```diff
@@ -120,7 +120,7 @@ export class Null extends Value(Object, IonTypes.NULL, FromJsConstructor.NONE) {
     this._unsupportedOperationOrNullDereference("fieldNames");
   }
 
-  fields(): [string, Value][] {
+  fields(): [string, Value[]][] {
```

Comment:
Unfortunately, we can't change this method signature, the `Struct` constructor's signature, or how iteration works as those would be breaking changes requiring a new major version.

I think we should leave those as-is and add new functions that allow users to opt-in to the array format. For example:

```ts
.get(name)
// vs
.getAll(name)

.fields()
// vs
.allFields()

for (let [fieldName, fieldValue] of s) {/*...*/}
// vs
for (let [fieldName, fieldValueList] of s.allFields()) {/*...*/}
```

While I think `get()`/`getAll()` is a clear single-value/list-of-values name pair, I'm less enthusiastic about `fields()`/`allFields()`. Open to suggestions there. `fields()`/`fieldsWithAllValues` is more accurate but pretty verbose.

Iteration is interesting, because it's _possible_ to support multiple fields given the existing signatures. For example, the constructor takes an iterator:

```ts
constructor(fields: Iterable<[string, Value]>, annotations: string[] = []) { /*...*/ }
```

The `fields` iterable could yield `["foo", 4]` and then `["foo", 6]`. I think the constructor can/should support that, with a later call to `.getAll("foo")` returning `[4, 6]`.

On the other hand, I _don't_ think `Struct`'s iterator should produce the same behavior because that would be a breaking change. Some number of folks may already be relying on deduplication in their iteration. I think `Struct`'s iterator should stay the same; users interested in all values can iterate over `myStruct.allFields()` (or whatever we name it) instead.

Reply:
I suppose the constructor could also accept a `Iterable<[string, Value]> | Iterable<[string, Value[]]>` to be more flexible.